File: src/main/java/de/tectoast/emolga/commands/flo/RemindCommand.java
package de.tectoast.emolga.commands.flo;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.utils.Constants;
import de.tectoast.emolga.utils.sql.DBManagers;
import net.dv8tion.jda.api.entities.TextChannel;

import java.sql.Timestamp;

import static de.tectoast.emolga.utils.Constants.CALENDAR_MSGID;
import static de.tectoast.emolga.utils.Constants.CALENDAR_TCID;

public class RemindCommand extends Command {

    public RemindCommand() {
        super("remind", "Setzt einen Reminder auf", CommandCategory.Flo, Constants.MYSERVER);
    }

    @Override
    public void process(GuildCommandEvent e) {
        try {
            String[] split = WHITESPACES_SPLITTER.split(e.getMessage().getContentRaw(), 3);
            long expires = parseCalendarTime(split[1]);
            String message = split[2];
            TextChannel calendarTc = e.getJDA().getTextChannelById(CALENDAR_TCID);
            DBManagers.CALENDAR.insertNewEntry(message, new Timestamp(expires / 1000 * 1000));
            scheduleCalendarEntry(expires, message);
            e.getMessage().delete().queue();
            calendarTc.editMessageById(CALENDAR_MSGID, buildCalendar()).queue();
        } catch (NumberFormatException ex) {
            e.getChannel().sendMessage("Das ist keine valide Zeitangabe!").queue();
            ex.printStackTrace();
        } catch (Exception ex) {
            e.getChannel().sendMessage("Es ist ein Fehler aufgetreten!").queue();
            ex.printStackTrace();
        }
    }


}

File: src/main/kotlin/de/tectoast/emolga/commands/flo/RemindCommand.kt
package de.tectoast.emolga.commands.flo

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.utils.Constants
import de.tectoast.emolga.utils.Constants.CALENDAR_MSGID
import de.tectoast.emolga.utils.Constants.CALENDAR_TCID
import de.tectoast.emolga.utils.sql.managers.CalendarManager
import java.sql.Timestamp

class RemindCommand : Command("remind", "Setzt einen Reminder auf", CommandCategory.Flo, Constants.G.MY) {
    override suspend fun process(e: GuildCommandEvent) {
        try {
            val args = e.arguments
            val expires = parseCalendarTime(args.getText("date"))
            val message = args.getText("text")
            CalendarManager.insertNewEntry(message, Timestamp(expires / 1000 * 1000))
            scheduleCalendarEntry(expires, message)
            e.reply("Reminder gesetzt!", ephermal = true)
            e.jda.getTextChannelById(CALENDAR_TCID)!!.editMessageById(CALENDAR_MSGID, buildCalendar()).queue()
        } catch (ex: NumberFormatException) {
            e.reply("Das ist keine valide Zeitangabe!", ephermal = true)
            ex.printStackTrace()
        } catch (ex: Exception) {
            e.reply("Es ist ein Fehler aufgetreten!", ephermal = true)
            ex.printStackTrace()
        }
    }

    init {
        argumentTemplate = ArgumentManagerTemplate.builder()
            .add("text", "Text", "Der Text", ArgumentManagerTemplate.Text.any())
            .add("date", "Datum", "Das Datum", ArgumentManagerTemplate.Text.any())
            .build()
        slash()
    }
}
